src/fdb5/api/FDB.h

```diff
@@ -72,73 +76,200 @@ class FDB {
+    /// @param handle eckit::message::Message to data to archive
     void archive(eckit::message::Message msg);
+
+    /// Archive binary data to a FDB.
+    /// Reads messages from the eckit::DataHandle and calls archive() the the
+    /// corresponding messages.
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param handle eckit::DataHandle reference data to archive
     void archive(eckit::DataHandle& handle);
+
+    /// Archive binary data to a FDB.
+    /// Internally creates a DataHandle and calls archive().
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param data Pointer to the binary data to archive
+    /// @param length Size of the data to archive with the given
     void archive(const void* data, size_t length);
-    // warning: not high-perf API - makes sure that all the requested fields are archived and there are no data
-    // exceeding the request
+
+    /// Archives data from Datahandle and ensures all keys exactly match the provided MarsRequest.
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param request a mars request
+    /// @param handle a data handle pointing to the data
+    /// @throws eckit::UserError if there are more keys in the MarsRequest then in the messages.
+    /// @throws eckit::UserError if message key not present in MarsRequest.
     void archive(const metkit::mars::MarsRequest& request, eckit::DataHandle& handle);
 
-    // disclaimer: this is a low-level API. The provided key and the corresponding data are not checked for consistency
-    // Optional callback function is called upon receiving field location from the store.
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @note No constistency checks are applied. The caller needs to ensure the provided key matches metadata present
+    /// in data.
+    /// @param key Key used for indexing and archiving the data
+    /// @param data Pointer to the binary data to archive
+    /// @param length Size of the data to archive with the given @p key
     void archive(const Key& key, const void* data, size_t length);
 
+    /// Generate an new index entry for an existing field location.
+    /// Can be used to reindex existing data into a new catalogue (see fdb-reindex tool).
+    /// @param key Key used to index the data.
+    /// @param location Location of existing data in an FDB store.
     void reindex(const Key& key, const FieldLocation& location);
 
-    /// Flushes all buffers and closes all data handles into a consistent DB state
+    /// Flush all buffers and closes all data handles into a consistent DB state
     /// @note always safe to call
     void flush();
 
+    /// Read binary data from an URI.
+    /// @param uri eckit uri to the data source
+    /// @return DataHandle for reading the requested data from
```

Comment:
Add note to review. This is a bit odd. The purpose of a URI is that it directly describes the data locations - and as such shouldn't need the FDB object to do the conversion into DataHandle?

Reply:
done